service/collector/client.go

```diff
@@ -0,0 +1,47 @@
+package collector
+
+import (
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/credential"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// credentialNamespace is the namespace in which we store credentials.
+	credentialNamespace = "giantswarm"
+
+	// credentialLabelSelector is the label selector we use to retrieve credentials.
+	credentialLabelSelector = "app=credentiald"
+)
+
+// getClientSets fetches all unique Azure clients.
```

Comment:
hey @JosephSalisbury , I know a lot of time has passed but we are getting some errors

![image](https://user-images.githubusercontent.com/627038/76417764-bf605c00-639d-11ea-9408-811b1378d08d.png)

and I suspect it has to do with collecting the same resource groups multiple times. I guess it could be that different secrets contain the same data (subscription id, tenant id, etc), so they are appended to the client list, even though they are effectively the same client. Do you think this is a fair assessment? Could different clients see the same resource groups? I don't think so, since they would be using different subscriptions.

Reply:
I'm checking the installation where this is happening. We have two secrets with different client id's, but they use the same subscription id. Could that be the problem? Two different clients can see the same resource groups since they use the same subscription.

Reply:
https://github.com/giantswarm/azure-operator/pull/713